Project cards 51

vsheo opened the pull request:
## What does this change?

Resolves user strory #12 

<!-- A PR should have enough detail to be understandable far in the future. e.g what is the problem/why is the change needed, how does it solve it and any questions or points of discussion. Prefer copying information from a GitHub issue over linking to it; the card may not always exist and reviewers may not have access to the board. -->

For this user story, I created two components: one for the project cards and a second for the container that holds them. They currently look like this:

[Untitled Video - Made With Clipchamp (2).webm](https://github.com/user-attachments/assets/952e5c3d-2903-4564-b613-fba20d8872b9)

## How Has This Been Tested?
<!-- Link to test results in the Wiki-->

- [ ] [User test]()
- [x] [Accessibility test]()
- [ ] [Performance test]()
- [x] [Responsive Design test]()
- [x] [Device test]()
- [x] [Browser test]()

All my test were done in this issue #63 

## How to review

<!-- Provide instructions to help others verify the change. This could take the form of "On PROD, do X and witness Y. On this branch, do X and witness Z. " -->

If you open this branch and start the localhost, you’ll be able to see my components as soon as you click the link.
I would like your feedback on my code:
- Am I following proper coding conventions?
- Is it clear and understandable?

I would like your feedback on the design:
- Does anything look off?
- Could anything be improved?
- Did I forget anything?

vsheo committed:
feat: HTML for the project cards #51

vsheo committed:
Merge branch 'styleguide-fonts-35' into project-cards-51

vsheo committed:
feat: mobile card styling #51

vsheo committed:
feat: font fixes #51

vsheo committed:
fix: naming according to conventions #52

vsheo committed:
feat: responsive container #52

vsheo committed:
feat: 2nd desktop column #52

vsheo committed:
fix: columns for big screens #52

vsheo committed:
Merge branch 'dev' into project-cards-51

vsheo committed:
feat: pleasurable details #64

vsheo committed:
fix: html validator #63

vsheo committed:
fix: responsiveness #63

vsheo committed:
fix: tab index #63

vsheo committed:
refactor: removed items

removed everything that should not merge with the dev branch

vsheo committed:
feat: dynamic data #73

vsheo committed:
feat: filter tags in project cards #73

Pink-Mallow commented:
## Feedback on code:

### Am I following proper coding conventions?
_You are following the right coding conventions_

### Is it clear and understandable?
_The code looks neat and in order. You have thought of fallbacks in case something is missing with the Else-if blocks for the projects cards._

## Feedback on the design:

### Does anything look off?
_Yes. It is a small detail though. There isn't much breathing space in the contents of Cards showing the Projects. For example: the text is too close to the edge of the cards and a bit close to the filters. Give it some more padding._

### Could anything be improved?
_Two things:_
* _Give the text some breathing space from the image, filters and from the border of the Card._
* _Be consistent with the shapes._
_The rest is looking like a good start._

_**This is the current styling of the Card:**_

  <img width="256" height="702" alt="image" src="https://github.com/user-attachments/assets/0c3c1ac1-bfe1-4b03-8193-03e7c19f056a" />
<br>    
<br>    

**My advice Instead, do this:**  
* _The description won't look like it is so close the border of the card and the filters. It is even in line with the title (i later on realized it's only for mobile and not exactly in line once it gets to desktop. But i hop you understand what i mean 🙏._

    <img width="256" height="680" alt="image" src="https://github.com/user-attachments/assets/5363885a-707c-4873-95c2-2cd0af7c7315" />
<br>
<br>

**Also a minuscule detail,**  
_Be **consistent** on the border radius so that the forms look cleaning and fitting. For example:_

  <img width="250" height="714" alt="image" src="https://github.com/user-attachments/assets/28055be5-df29-49f7-9794-819659dcca50" />
<br>  
_Your card is in `border-radius= 15px;` and to make it cleaner over the rest of the elements, half the size of other radius so that it feels like the corners fit nicely with the card itself by giving the `img` and `li` a `border-radius: 7.5px` instead._ 
<br>

**Instead of:**  
<img width="250" height="748" alt="image" src="https://github.com/user-attachments/assets/9f43678d-75cc-45a0-bd47-fdb86f5ca919" />
<br>
<br>

### Did I forget anything?
_So far you aren't missing anything You're on the right track. It is just the small details that can be looked at later on._

Pink-Mallow reviewed (changes requested):
On the designing part:
- **Border-radius:**
   - Change the border radius to a more snug fit  by giving `<img>` and `<li>` a radius of `7.5px` as the radius of the card (`<article>`) is `15px`.
  
- **Text:**
   - Give the the description some space in the <ins>Card</ins> to be more inline with the left side from the title, and more spacing from the image (desktop version), and finally from the filters (in mobile version).
   - The Title in the cards can have more space from the top of the card (desktop version) with `marging` top or `padding`).

vsheo committed:
style: added comment to index.js #65

vsheo commented:
@Mikiyas-hs 
I’ve added more comments in index.js, so it’s now clear what each component was created for.

The color is different because it now matches the color used for this section on https://humanvaluesforsmartercities.nl/
In the previous sprint review, the product owner mentioned that the colors should better align with that site.
The color `--oklch-neutral` is almost the same as what we see on that website.  
[Here’s an example](https://github.com/fdnd-agency/designingforrecognition/issues/35#issuecomment-3387274141), the hex version is what is used on https://humanvaluesforsmartercities.nl/
the lighter and darker variations are based on that color.

Regarding the commits: since the branch for styleguide.css hadn’t been reviewed yet / wasn’t on the dev branch, I first merged it into this branch so I could use the CSS for this component.  
This caused all the commits from that branch to be included in this one.

I overlooked the gradient when creating this component.  
Here’s a video showing how it looks when it matches the Figma design.

[Gradient-V1.webm](https://github.com/user-attachments/assets/8811f1d1-4caa-4af7-a4fd-3b5751b30f48)

In this version, only the last part has a gradient.  

[Gradient-V2.webm](https://github.com/user-attachments/assets/08021f64-909c-4039-9052-28d3f281e6b0)

Which do you think looks better?
In my opinion, it would be better to have most of this section in dark blue and then transition to white at the bottom. This makes it feel like the project cards belong to the same section, and only change when a new section starts.

vsheo committed:
style: project cards padding and margins #65

vsheo commented:
@Pink-Mallow

I added more padding to the description and also adjusted the margins so that the title, date, and description all start at the same position.

After that, I adjusted the border radius so that all child elements have half the radius of the parent (in this case, the article).  
This does indeed look better.

Are the margins and padding okay now?

vsheo committed:
feat: linear gradient background #65
